src/main/scala/mesosphere/jackson/CaseClassValueInstantiator.scala

```diff
@@ -64,7 +64,9 @@ protected class CaseClassValueInstantiator(
     ListMap(applySymbol.paramss.flatten.zipWithIndex.map {
       case (p, i) =>
         val typeSig = p.asTerm.typeSignature.typeSymbol.asClass
-        val cls = classLoaderMirror.runtimeClass(typeSig)
+        if (typeSig.fullName != "scala.Array") {
+          classLoaderMirror.runtimeClass(typeSig)
```

Comment:
Is it really necessary to leave the method call here? It seems we could write just
```
    ListMap(applySymbol.paramss.flatten.zipWithIndex.map {
      case (p, i) =>
        p.name.toString -> valueFor(i)
    }: _*)
```